### PR TITLE
fix: get_queryset sends self in fields.py which it shouldnt

### PR DESCRIPTION
**Pull request by deshk04**
details of the issue can be found at https://github.com/strawberry-graphql/strawberry-graphql-django/issues/187 test cases are also update

## Description

Details for the issue can be found at https://github.com/strawberry-graphql/strawberry-graphql-django/issues/187

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

https://github.com/strawberry-graphql/strawberry-graphql-django/issues/187

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tests/test_pagination.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Fruit:
 class BerryFruit:
     name: auto
 
+    @classmethod
     def get_queryset(self, queryset, info, **kwargs):
```
Comment (bellini666) [inline tests/test_pagination.py:22]:
```suggestion
    def get_queryset(cls, queryset, info, **kwargs):
```

File: tests/test_queries.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ class BerryFruit:
     name_upper: str
     name_lower: str
 
+    @classmethod
     def get_queryset(self, queryset, info):
```
Comment (bellini666) [inline tests/test_queries.py:40]:
```suggestion
    def get_queryset(cls, queryset, info):
```

File: tests/test_queries.py [hunk 3]
```diff
@@ -48,6 +49,7 @@ class FruitInterface:
 
 @strawberry_django.type(models.Fruit)
 class BananaFruit(FruitInterface):
+    @classmethod
     def get_queryset(self, queryset, info):
```
Comment (bellini666) [inline tests/test_queries.py:53]:
```suggestion
    def get_queryset(cls, queryset, info):
```

**Review by bellini666 — CHANGES_REQUESTED** [review]
Just a couple of fixes to the documentation and this should be good to merge :)

**Review by bellini666 — APPROVED** [review]
Thank you!

**Comment by fireteam99** [comment]
Opened a PR to update the documentation regarding this change: https://github.com/strawberry-graphql/strawberry-graphql-django/pull/199

**Pull request merged**